crates/turbo-tasks-memory/src/task.rs

```diff
@@ -1752,14 +1825,21 @@ impl Task {
     ) {
         let mut state = self.full_state_mut();
         if state.children.insert(child_id) {
+            if let TaskScopes::Inner(_, optimization_counter) = &state.scopes {
+                if *optimization_counter * state.children.len() > SCOPE_OPTIMIZATION_THRESHOLD {
```

Comment:
This check could be a function (reused above):

```rust
should_add_to_scope(optimization_counter, children_count)
```

Fewer opportunities for error, and lets you inline the const in there and comment the function directly.